### PR TITLE
Goog signin

### PR DESCRIPTION
**Pull request by Aliblas**
Implements the sign in button, with an example provided in test.jsp. Also includes frontend getId() method.
Merges with origin may have been a mistake...

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/UserStatusServlet.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     JsonObject userInfo = new JsonObject();
     userInfo.addProperty("isUserLoggedIn", UserHelper.isUserLoggedIn());
     if (UserHelper.isUserLoggedIn()) {
-        System.out.println("User is logged in");
+        System.out.println("Return user profile info");
```
Comment (isamitre) [inline src/main/java/com/google/sps/servlets/UserStatusServlet.java]: why'd you change what it prints out? doesn't the original make more sense?

File: src/main/webapp/css/style-sign-in.css [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#sign-in-box {
+    border-radius: 5px;
```
Comment (isamitre) [inline src/main/webapp/css/style-sign-in.css]: nit: everything should have +2 spacing

File: src/main/webapp/css/style-sign-in.css [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    vertical-align: middle;
+    width: 48px;
+}
+
+#profile-photo {
+    border-radius: 50%;
+    height: 100%;
+    object-fit: cover;
+    width: 100%;
+    vertical-align: middle;
```
Comment (isamitre) [inline src/main/webapp/css/style-sign-in.css]: should be alphabetical

File: src/main/webapp/javascript/tests.js [hunk 4]
```diff
@@ -142,4 +142,15 @@ testRecipeCard = () => {
...
   return;
+}
+
+/**
+ * Test for confirmUser method in user-auth.js
+ * This function is called onclick of <sign in required service> button in test.jsp
+ */
+function userSpecificService() {
+    confirmUser().then((googleUser) => {
+        outputPlace = document.getElementById('test-output');
```
Comment (isamitre) [inline src/main/webapp/javascript/tests.js]: nit: +2 spacing

File: src/main/webapp/javascript/user-auth.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
+var auth2;
+
+function getId() {
+    return auth2.currentUser.get().getId();
```
Comment (isamitre) [inline src/main/webapp/javascript/user-auth.js]: nit: +2 spacing

File: src/main/webapp/javascript/user-auth.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+function onStart() {
+  console.log("onstart");
+  gapi.load('auth2', initSigninV2);
```
Comment (isamitre) [inline src/main/webapp/javascript/user-auth.js:14]: a comment explaining what this function does would be helpful

**Review by isamitre — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/UserStatusServlet.java (hunk 1 above)
Comment (Aliblas) [reply src/main/java/com/google/sps/servlets/UserStatusServlet.java]: I'm actually gonna delete this whole file so I won't think about this question too much.

File: src/main/webapp/css/style-sign-in.css [hunk 7]
```diff
@@ -0,0 +1,57 @@
+#sign-in-box {
+  border-radius: 5px;
+  border: 1px solid grey;
```
Comment (AlexanderEllis) [inline src/main/webapp/css/style-sign-in.css:3]: Super nit pick: `border` should come before `border-radius`.

File: src/main/webapp/javascript/user-auth.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  return auth2.currentUser.get().getId();
+}
+
+/**
+ * loads the auth2 library for the g-signin API, and then initializes
+ * the googleAuth object.
+ * passed as jquery parameter, executed by API script.
+ */
+function onStart() {
+  console.log("onstart");
```
Comment (AlexanderEllis) [inline src/main/webapp/javascript/user-auth.js:13]: Are these `console.log` statements just for your debugging? If so, you can remove them when merging into master.

File: src/main/webapp/javascript/user-auth.js [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+  gapi.load('auth2', initSigninV2);
+}
+
+/**
+ * initializes google authentication API and sets event listeners.
+ */
+function initSigninV2() {
+  auth2 = gapi.auth2.init({
+    client_id : '1034390229233-u07o0iaas2oql8l4jhe7fevpfsbrtv7n.apps.googleusercontent.com'
```
Comment (AlexanderEllis) [inline src/main/webapp/javascript/user-auth.js:22]: Do you maybe want to make this client_id a constant at the top of the file with a comment on where it comes from?

File: src/main/webapp/javascript/user-auth.js [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+    if (signedIn) {
+      console.log('enable dropdown');
+      enableUserDropdown();
+    } else {
+      console.log('disable dropdown');
+      disableUserDropdown();
+    }
+  });
+}
+
```
Comment (AlexanderEllis) [inline src/main/webapp/javascript/user-auth.js:41]: Super nit: I think you have two extra blank lines here.

File: src/main/webapp/javascript/user-auth.js [hunk 11]
```diff
@@ -0,0 +1,96 @@
+var auth2;
+
+function getId() {
+  return auth2.currentUser.get().getId();
+}
+
+/**
+ * loads the auth2 library for the g-signin API, and then initializes
```
Comment (AlexanderEllis) [inline src/main/webapp/javascript/user-auth.js:8]: Super nit: capitalize the first word in each of the function comment sentence (here and other comments in the file).

File: src/main/webapp/css/style-sign-in.css [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+.dropdown {
+  cursor: pointer;
+  display: inline-block;
+  position: relative;
+}
+
+.dropdown-content {
+  background-color: #f9f9f9;
+  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
```
Comment (AlexanderEllis) [inline src/main/webapp/css/style-sign-in.css:40]: Super nit: if you're using zero pixels (for something other than the `flex` property), you don't need the `px`. https://google.github.io/styleguide/htmlcssguide.html#0_and_Units

Also please add some spaces to your rgba, e.g. `rgba(0, 0, 0, 0.2)`.

File: src/main/webapp/javascript/user-auth.js [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+/**
+ * To be run in currentUser listener, activates whenever current user changes.
+ *
+ * @param {GoogleUser} current user to provide profile info.
+ */
+function insertUserInfo(googleUser) {
+  // insert user name into user-page anchor
+  let text = googleUser.getBasicProfile().getGivenName() + '\'s Page';
```
Comment (AlexanderEllis) [inline src/main/webapp/javascript/user-auth.js:60]: Does this still create an error if the user isn't initially signed in like we saw yesterday? If so, you may want to do a check on `googleUser` and `googleUser.getBasicProfile()`.

**Review by AlexanderEllis — APPROVED** [review]
Looks good to me! Just a few minor nits and one question about the `getGivenName` error we saw yesterday.

Would you also mind adding a screenshot or screencast to the PR description? That's usually super helpful for seeing what has changed visually.

**Pull request merged**